.pre-commit-config.yaml

```
@@ -0,0 +1,44 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+      - id: check-toml
+        alias: toml
+      - id: check-yaml
+        alias: yaml
+      - id: end-of-file-fixer
+        alias: eof
+      - id: trailing-whitespace
+        alias: whitespace
+
+  - repo: https://github.com/pre-commit/pygrep-hooks
+    rev: v1.9.0
+    hooks:
+      - name: Prevent blanket NOQAs
+        id: python-check-blanket-noqa
+        alias: noqa
+      - name: Prevent blanket type ignores
+        id: python-check-blanket-type-ignore
+        alias: type-ignore
+
+  - repo: https://github.com/psf/black
+    rev: 21.10b0
+    hooks:
+      - id: black
+        language_version: python3
+
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.8.0
+    hooks:
+      - id: isort
+        language_version: python3
+
+  - repo: local
+    hooks:
+      - name: Flake8
+        id: flake8
+        description: Runs Flake8 within the project's environment
+        entry: poetry run flake8 --config=tox.ini --diff
```

```suggestion
        entry: poetry run flake8
```
Pretty sure these aren't needed as flake8 will find the tox file itself, and pre-commit only passes the diff to flake8 too.